lmdeploy/vl/utils.py

```diff
@@ -45,27 +53,35 @@ def load_image_from_base64(image: Union[bytes, str]) -> Image.Image:
 
 def load_image(image_url: Union[str, Image.Image]) -> Image.Image:
     """load image from url, local path or openai GPT4V."""
-    if isinstance(image_url, Image.Image):
-        return image_url
-
     FETCH_TIMEOUT = int(os.environ.get('LMDEPLOY_FETCH_TIMEOUT', 10))
     headers = {
         'User-Agent':
         'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 '
         '(KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'
     }
-    if image_url.startswith('http'):
-        response = requests.get(image_url,
-                                headers=headers,
-                                timeout=FETCH_TIMEOUT)
-        response.raise_for_status()
+    try:
+        ImageFile.LOAD_TRUNCATED_IMAGES = True
+        if isinstance(image_url, Image.Image):
+            img = image_url
+        elif image_url.startswith('http'):
+            response = requests.get(image_url,
+                                    headers=headers,
+                                    timeout=FETCH_TIMEOUT)
+            response.raise_for_status()
+            img = Image.open(BytesIO(response.content))
+        elif image_url.startswith('data:image'):
+            img = load_image_from_base64(image_url.split(',')[1])
+        else:
+            # Load image from local path
+            img = Image.open(image_url)
 
-        # Open the image using PIL
-        img = Image.open(BytesIO(response.content))
-    elif image_url.startswith('data:image'):
-        img = load_image_from_base64(image_url.split(',')[1])
-    else:
-        # Load image from local path
-        img = Image.open(image_url)
+        # check image valid
+        img = img.convert('RGB')
```

Comment:
Is there a better way to validate the image?

Reply:
Currently all models will convert the image to RGB, and checking here will not increase the time consumption

Reply:
> Is there a better way to validate the image?

I'm not sure if `load()` function satisfied the need. Besides, if `convert('RGB')` is a must for all the model, we may do it here and remove them from `forward` function.